### PR TITLE
Add dark mode

### PR DESCRIPTION
**Pull request by benny-n**

### CODE REVIEW

File: packages/sogrim-app/src/components/App/UserApp.tsx [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+  const theme = useTheme();
+  const colorMode = React.useContext(ColorModeContext);
+
   useEffect(() => {
     if (!isLoading && data) {
       updateStoreUserDetails(data.details);
       updateStoreUserSettings(data.settings);
+      if (data.settings.dark_mode && theme.palette.mode !== DARK_MODE_THEME) {
+        colorMode.toggleColorMode();
+      }
```
Comment (liadaram1) [inline packages/sogrim-app/src/components/App/UserApp.tsx:29]: I think there might be a bug here.
What will happen in the following scenario:
The user changes in his device to dark mode. Now the value of `data.settings.dark_mode` is `true`.
Next, he is connecting from a different device, and changes back to `light mode`.
IMHO there will be a limbo state, where his original device will still be dark mode and his second device light mode.

**Review by liadaram1 — CHANGES_REQUESTED** [review]

File: packages/server/src/resources/user.rs [hunk 2]
```diff
@@ -12,6 +12,7 @@ pub struct UserDetails {
 #[derive(Default, Clone, Debug, Deserialize, Serialize)]
```
Comment (liadaram1) [inline packages/server/src/resources/user.rs:12]: In addition, I tried to use the `dark_mode` locally, and there is an annoying state where after changing the mode, and refresh the page we get the pop up "סטטוס התואר שלך לא מעודכן...".

**Review by liadaram1 — CHANGES_REQUESTED** [review]

File: packages/sogrim-app/src/components/App/UserApp.tsx (hunk 1 above)
Comment (liadaram1) [reply packages/sogrim-app/src/components/App/UserApp.tsx:29]: Ok I found a way to test it locally and it seems I am wrong :).
Closing the comment.

**Comment by benny-n** [comment]
This (breaking) change requires us to run 3 queries on the next release:
1. Update dark mode option: 
```js
db.Users.update({}, {$set: {"settings.dark_mode": true}})
```
2. Remove `compute_in_progress` from `UserSettings`: 
```js
db.Users.update({}, {$unset: {"settings.compute_in_progress": ""}})
```
3. Add `compute_in_progress` to `UserDetails`: 
```js
db.Users.update({}, {$set: {"settings.compute_in_progress": false}})
```

**Review by liadaram1 — APPROVED** [review]

**Pull request merged**